Adding troubleshooting tips and command line execution info to profiling.md (#2927)
JuliaGPU/CUDA.jl: docs/src/development/profiling.md

!["NVIDIA Nsight Compute - API inspection"](nsight_compute-api.png)

> Note that command line execution of Julia scripts also works. Run `ncu julia myscript.jl` without `--mode=launch` to profile the entire script. Use the ncu command linge arguments if desired.

Reviewer: ```suggestion
Note that command line execution of Julia scripts also works. Run `ncu julia myscript.jl` without `--mode=launch` to profile the entire script. Use the `ncu` command line arguments if desired. 
```

Is there a link to the ncu docs we could provide here?

> Note that command line execution of Julia scripts also works. Run `ncu julia myscript.jl` without `--mode=launch` to profile the entire script. Use the `ncu` command line arguments if desired. Refer to https://docs.nvidia.com/nsight-compute/NsightComputeCli/index.html for detailed documentation.

Reviewer 2: ```suggestion
Note that command line execution of Julia scripts also works. Run `ncu julia myscript.jl` without `--mode=launch` to profile the entire script. Use the `ncu` command line arguments if desired. Refer to [the NVIDIA documentation](https://docs.nvidia.com/nsight-compute/NsightComputeCli/index.html) for more details.
```

Make sure `Break On API Error` is disabled in the `Debug` menu, as CUDA.jl purposefully
triggers some API errors as part of its normal operation.

##### NaN values as output in NSight Compute.

> Incompatibility between ncu and julia CUDA version. Run `ncu --version` to find which CUDA toolkit is being used. In Julia set `CUDA.set_runtime_version!(v"xx.x")` to the same version. This version may be lower than the driver version output in `nvidia-smi`, but not higher.

Reviewer: ```suggestion
Incompatibility between the ncu and Julia CUDA version. Run `ncu --version` to find which CUDA toolkit is being used. In Julia set `CUDA.set_runtime_version!(v"xx.x")` to the same version. This version may be lower than the driver version output in `nvidia-smi`, but not higher.
```
